conans/client/cmake.py

```diff
@@ -123,12 +123,18 @@ def flags(self):
             flags.append('-DCONAN_COMPILER="%s"' % comp)
         if comp_version:
             flags.append('-DCONAN_COMPILER_VERSION="%s"' % comp_version)
-        if arch == "x86":
-            if op_system == "Linux":
+
+        if op_system == "Linux" or op_system == "FreeBSD" or op_system == "SunOS":
+            if arch == "x86":
                 flags.extend(["-DCONAN_CXX_FLAGS=-m32",
                               "-DCONAN_SHARED_LINKER_FLAGS=-m32",
                               "-DCONAN_C_FLAGS=-m32"])
-            elif op_system == "Macos":
+            if arch == "x86_64":
+                flags.extend(["-DCONAN_CXX_FLAGS=-m64",
```

Comment:
That depends on your platform, actually.
On BSD, Linux (and I suspect, Mac too), the compiler defaults to 64 bit binaries.
On Solaris, it defaults to 32 bit event on 64 bit systems.
But then again, it shouldn't hurt since if you ask for a x86_64 build you do want a 64 bit binary.